include/h2o/balancer.h

```diff
@@ -46,7 +57,8 @@ struct st_h2o_balancer_t {
 h2o_balancer_t *h2o_balancer_create_rr(void);
 
 /* least connection */
-h2o_balancer_t *h2o_balancer_create_lc(void);
+typedef void (*h2o_balancer_lc_get_conn_count_cb)(size_t *conn_count, h2o_balancer_backend_t **backends, size_t backends_len);
+h2o_balancer_t *h2o_balancer_create_lc(h2o_balancer_lc_get_conn_count_cb conn_count_cb);
```

Comment:
How about moving `least_count` into the balancer logic, and calling functions to increment / decrement the counters from socketpool?

The reason I suggest this is due to following reasons:
* leased_count is only used by the least-conn load balancer
* the counters are shared between threads and therefore updating them could have impact on performance; I'd prefer counting the number of connections only when necessary